low-level/sensors/impl/generic/raid_integrity_data.py

```diff
@@ -166,22 +169,37 @@ def _raid_health_monitor(self):
                 # Check RAID device array state is 'idle' or not
                 result = self._check_raid_state(device)
                 if result == "failed":
+                    logger.warning("'Idle' state not found for RAID device:{}"
+                                    .format(device))
+                    # Retry to check RAID state
                     self._retry_execution(self._check_raid_state, device)
-                logger.debug("'idle' state is found in Raid device:{}."
+                logger.info("'idle' state is found in Raid device:{}."
                              .format(device))
 
                 # Check Mismatch count in RAID device files.
                 result = self._check_mismatch_count(device)
                 if result == "failed":
+                    # Persist RAID device fault state and send alert
+                    fault_status_file = self.DEFAULT_RAID_DATA_PATH + device + "_" + RaidDataConfig.RAID_MISMATCH_FAULT_STATUS.value
+                    if os.path.exists(fault_status_file):
+                        with open(fault_status_file, 'r') as fs:
+                            data = fs.read().rstrip()
+                        if self.FAULT_RESOLVED in data:
+                            self._update_fault_state_file(device, self.FAULT, fault_status_file)
```

Comment:
Please move the file update to after sending the alert i.e. line #191.

Reply:
Done